From c478e50e023e909c22494bc6ffad33a080cedf2b Mon Sep 17 00:00:00 2001
From: Kito Cheng <kito.cheng@sifive.com>
Date: Mon, 11 May 2020 17:51:03 +0800
Subject: [PATCH 01/14] testsuite: Require gnu-tm support for pr94856.C

---
 gcc/testsuite/g++.dg/ipa/pr94856.C | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/testsuite/g++.dg/ipa/pr94856.C b/gcc/testsuite/g++.dg/ipa/pr94856.C
index 5315c52d80e..40f3a167e29 100644
--- a/gcc/testsuite/g++.dg/ipa/pr94856.C
+++ b/gcc/testsuite/g++.dg/ipa/pr94856.C
@@ -1,5 +1,6 @@
 /* { dg-do compile } */
 /* { dg-options "-O2 -fno-tree-dse --param uninlined-function-insns=0 --param early-inlining-insns=3 -fgnu-tm " } */
+/* { dg-require-effective-target fgnu_tm } */
 
 class a {
 public:
-- 
2.25.4

