
newlib-2.4.0 regression.
Export the prototype of gettimeofday

Signed-off-by: Juro Bystricky <juro.bystricky@intel.com>

--- a/newlib/libc/include/sys/time.h	2016-03-29 14:33:42.000000000 -0700
+++ b/newlib/libc/include/sys/time.h	2016-10-12 09:48:03.873642589 -0700
@@ -429,9 +429,10 @@
 int _EXFUN(getitimer, (int __which, struct itimerval *__value));
 int _EXFUN(setitimer, (int __which, const struct itimerval *__restrict __value,
 					struct itimerval *__restrict __ovalue));
+#endif
+
 int _EXFUN(gettimeofday, (struct timeval *__restrict __p,
 			  void *__restrict __tz));
-#endif
 
 #if __GNU_VISIBLE
 int _EXFUN(futimesat, (int, const char *, const struct timeval [2]));
