--- original-newlib/newlib/configure.host
+++ newlib/newlib/configure.host
@@ -210,6 +210,10 @@ case "${host_cpu}" in
   mep)
 	machine_dir=mep
 	;;
+  riscv*)
+	machine_dir=riscv
+	newlib_cflags="${newlib_cflags}"
+	;;
   mips*)
 	machine_dir=mips
 	;;
--- original-newlib/newlib/libc/include/machine/ieeefp.h
+++ newlib/newlib/libc/include/machine/ieeefp.h
@@ -170,6 +170,14 @@
 #define __IEEE_LITTLE_ENDIAN
 #endif
 
+#ifdef __riscv__
+# ifdef __RISCVEL__
+#  define __IEEE_LITTLE_ENDIAN
+# else
+#  define __IEEE_BIG_ENDIAN
+# endif
+#endif
+
 #ifdef __i960__
 #define __IEEE_LITTLE_ENDIAN
 #endif
--- original-newlib/newlib/libc/include/machine/setjmp.h
+++ newlib/newlib/libc/include/machine/setjmp.h
@@ -338,6 +338,11 @@ _BEGIN_STD_C
 #define _JBLEN 0x44
 #endif
 
+#ifdef __riscv__
+#define _JBTYPE long
+#define _JBLEN ((16*sizeof(long) + 12*sizeof(double))/sizeof(long))
+#endif
+
 #ifdef _JBLEN
 #ifdef _JBTYPE
 typedef	_JBTYPE jmp_buf[_JBLEN];
--- original-newlib/newlib/libc/machine/configure
+++ newlib/newlib/libc/machine/configure
@@ -11882,6 +11883,8 @@ subdirs="$subdirs a29k"
  ;;
 	mips) subdirs="$subdirs mips"
  ;;
+	riscv) subdirs="$subdirs riscv"
+ ;;
 	mn10200) subdirs="$subdirs mn10200"
  ;;
 	mn10300) subdirs="$subdirs mn10300"

