diff --git a/configure b/configure
index 18336376bf..07cd016280 100755
--- a/configure
+++ b/configure
@@ -1923,7 +1923,6 @@ if test "$skip_meson" = no; then
   echo "widl = [$(meson_quote $widl)]" >> $cross
   echo "windres = [$(meson_quote $windres)]" >> $cross
   echo "windmc = [$(meson_quote $windmc)]" >> $cross
-  if test "$cross_compile" = "yes"; then
     echo "[host_machine]" >> $cross
     echo "system = '$host_os'" >> $cross
     case "$cpu" in
@@ -1951,7 +1950,6 @@ if test "$skip_meson" = no; then
     mv $native config-meson.native
     meson_option_add --native-file
     meson_option_add config-meson.native
-  fi
   mv $cross config-meson.cross
   meson_add_machine_file config-meson.cross
   if test -f "$source_path/configs/meson/$host_os.txt"; then
