diff --git a/include/opcode/riscv-opc.h b/include/opcode/riscv-opc.h
index d10c7f8..4e398a5 100644
--- a/include/opcode/riscv-opc.h
+++ b/include/opcode/riscv-opc.h
@@ -225,6 +225,8 @@
 #define MASK_MRET  0xffffffff
 #define MATCH_DRET 0x7b200073
 #define MASK_DRET  0xffffffff
+#define MATCH_ERET 0x10000073
+#define MASK_ERET  0xffffffff
 #define MATCH_SFENCE_VM 0x10400073
 #define MASK_SFENCE_VM  0xfff07fff
 #define MATCH_WFI 0x10500073
@@ -814,6 +816,7 @@ DECLARE_INSN(sret, MATCH_SRET, MASK_SRET)
 DECLARE_INSN(hret, MATCH_HRET, MASK_HRET)
 DECLARE_INSN(mret, MATCH_MRET, MASK_MRET)
 DECLARE_INSN(dret, MATCH_DRET, MASK_DRET)
+DECLARE_INSN(eret, MATCH_ERET, MASK_ERET)
 DECLARE_INSN(sfence_vm, MATCH_SFENCE_VM, MASK_SFENCE_VM)
 DECLARE_INSN(wfi, MATCH_WFI, MASK_WFI)
 DECLARE_INSN(csrrw, MATCH_CSRRW, MASK_CSRRW)
diff --git a/opcodes/riscv-opc.c b/opcodes/riscv-opc.c
index c08f226..03dc690 100644
--- a/opcodes/riscv-opc.c
+++ b/opcodes/riscv-opc.c
@@ -610,6 +610,7 @@ const struct riscv_opcode riscv_opcodes[] =
 {"hret",      "I",   "",     MATCH_HRET, MASK_HRET, match_opcode, 0 },
 {"mret",      "I",   "",     MATCH_MRET, MASK_MRET, match_opcode, 0 },
 {"dret",      "I",   "",     MATCH_DRET, MASK_DRET, match_opcode, 0 },
+{"eret",      "I",   "",     MATCH_ERET, MASK_ERET, match_opcode, 0 },
 {"sfence.vm", "I",   "",     MATCH_SFENCE_VM, MASK_SFENCE_VM | MASK_RS1, match_opcode, 0 },
 {"sfence.vm", "I",   "s",    MATCH_SFENCE_VM, MASK_SFENCE_VM, match_opcode, 0 },
 {"wfi",       "I",   "",     MATCH_WFI, MASK_WFI, match_opcode, 0 },
