From 31fcf3bc76285ff5389b71842d8456ae14ac6312 Mon Sep 17 00:00:00 2001
From: Anas Nashif <anas.nashif@intel.com>
Date: Wed, 7 Aug 2019 05:23:08 -0500
Subject: [PATCH 10/17] nios2: add multilib support

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
---
 gcc/config/nios2/t-nios2 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gcc/config/nios2/t-nios2 b/gcc/config/nios2/t-nios2
index cfee4f82003..63b7499cb13 100644
--- a/gcc/config/nios2/t-nios2
+++ b/gcc/config/nios2/t-nios2
@@ -18,9 +18,9 @@
 # along with GCC; see the file COPYING3.  If not see
 # <http://www.gnu.org/licenses/>.
 
-# MULTILIB_OPTIONS = mno-hw-mul/mhw-mulx mcustom-fpu-cfg=60-1/mcustom-fpu-cfg=60-2
-# MULTILIB_DIRNAMES = nomul mulx fpu-60-1 fpu-60-2
-# MULTILIB_EXCEPTIONS = 
+MULTILIB_OPTIONS = mno-hw-mul/mhw-mulx mcustom-fpu-cfg=60-1/mcustom-fpu-cfg=60-2
+MULTILIB_DIRNAMES = nomul mulx fpu-60-1 fpu-60-2
+MULTILIB_EXCEPTIONS =
 
 # MULTILIB_OPTIONS += mel/meb
 # MULTILIB_DIRNAMES += le be
-- 
2.30.2

