--- /dev/null	2016-09-16 18:21:25.208077892 -0700
+++ b/gcc/config/i386/t-zephyr	2015-01-05 04:33:28.000000000 -0800
@@ -0,0 +1,23 @@
+# Copyright (C) 2016 Intel Corporation.
+#
+# This file is part of GCC.
+#
+# GCC is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# GCC is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GCC; see the file COPYING3.  If not see
+# <http://www.gnu.org/licenses/>.
+#
+
+MULTILIB_OPTIONS += msoft-float
+MULTILIB_DIRNAMES += soft-float
+MULTILIB_MATCHES = msoft-float=mno-80387
+MULTILIB_EXCEPTIONS = 

