diff --git a/meta/recipes-devtools/automake/automake_1.15.bb b/meta/recipes-devtools/automake/automake_1.15.1.bb
similarity index 80%
rename from meta/recipes-devtools/automake/automake_1.15.bb
rename to meta/recipes-devtools/automake/automake_1.15.1.bb
index a3c72fd334..11e1d10bb7 100644
--- a/meta/recipes-devtools/automake/automake_1.15.bb
+++ b/meta/recipes-devtools/automake/automake_1.15.1.bb
@@ -3,7 +3,7 @@ LICENSE = "GPLv2"
 LIC_FILES_CHKSUM = "file://COPYING;md5=751419260aa954499f7abaabaa882bbe"
 DEPENDS_class-native = "autoconf-native"

-NAMEVER = "${@oe.utils.trim_version("${PV}", 2)}"
+NAMEVER = "${@oe.utils.trim_version("${PV}", 3)}"

 RDEPENDS_${PN} += "\
     autoconf \
@@ -23,8 +23,8 @@ SRC_URI += " file://python-libdir.patch \
             file://performance.patch \
             file://new_rt_path_for_test-driver.patch"

-SRC_URI[md5sum] = "716946a105ca228ab545fc37a70df3a3"
-SRC_URI[sha256sum] = "7946e945a96e28152ba5a6beb0625ca715c6e32ac55f2e353ef54def0c8ed924"
+SRC_URI[md5sum] = "95df3f2d6eb8f81e70b8cb63a93c8853"
+SRC_URI[sha256sum] = "988e32527abe052307d21c8ca000aa238b914df363a617e38f4fb89f5abf6260"

 CACHED_CONFIGUREVARS += "ac_cv_path_PERL=${USRBINPATH}/perl"
