diff --git a/gcc/config/riscv/t-elf b/gcc/config/riscv/t-elf
index ebb6e92..095c8b15 100644
--- a/gcc/config/riscv/t-elf
+++ b/gcc/config/riscv/t-elf
@@ -1,9 +1,9 @@
 # Build the libraries for both hard and soft floating point
 
-MULTILIB_OPTIONS = m64/m32
-MULTILIB_DIRNAMES = 64 32
+#MULTILIB_OPTIONS = m64/m32
+#MULTILIB_DIRNAMES = 64 32
 
 ifneq ($(with_float), soft)
-MULTILIB_OPTIONS += mno-float
+MULTILIB_OPTIONS += mfloat-abi=soft
 MULTILIB_DIRNAMES += soft-float
 endif
