diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -62,9 +62,9 @@
 	threads="pthreads"
 
 	# HIDAPI/hidraw libs
-	PKG_CHECK_MODULES([libudev], [libudev], true, [hidapi_lib_error libudev])
-	LIBS_HIDRAW_PR+=" $libudev_LIBS"
-	CFLAGS_HIDRAW+=" $libudev_CFLAGS"
+#	PKG_CHECK_MODULES([libudev], [libudev], true, [hidapi_lib_error libudev])
+#	LIBS_HIDRAW_PR+=" $libudev_LIBS"
+#	CFLAGS_HIDRAW+=" $libudev_CFLAGS"
 
 	# HIDAPI/libusb libs
 	AC_CHECK_LIB([rt], [clock_gettime], [LIBS_LIBUSB_PRIVATE+=" -lrt"], [hidapi_lib_error librt])
